File: src/backend/java/org/example/Angular/entities/Item.java
package org.example.Angular.entities;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.HashSet;
import java.util.List;
import java.util.Set;

@Entity
public class Item {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    private String name;
    private String description;
    private double price;
    private int count;

    private int subcategory;

    @ElementCollection
    @CollectionTable(name = "image", joinColumns = @JoinColumn(name = "item_id"))
    @Column(name = "filename")
    private Set<String> images = new HashSet<>();

    @ElementCollection(fetch = FetchType.EAGER)
    private List<String> looks = new ArrayList<>();

//  Constructors

    public Item() { }

    public Item(String name, String description, double price, int count, int subcategory) {
        this.name = name;
        this.description = description;
        this.price = price;
        this.count = count;
        this.subcategory = subcategory;
    }

//  Getters & Setters

    public int getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public double getPrice() {
        return price;
    }

    public void setPrice(double price) {
        this.price = price;
    }

    public int getCount() {
        return count;
    }

    public void setCount(int count) {
        this.count = count;
    }

    public int getSubCategory() {
        return subcategory;
    }

    public void setSubCategory(int subcategory) {
        this.subcategory = subcategory;
    }

    public Set<String> getImages() {
        return images;
    }

    public void addImage(String name){
        images.add(name);
    }

    public List<String> getLooks() {
        return looks;
    }

    public void addLook(String ip){
        getLooks().add(ip);
    }
}

File: src/frontend/src/app/models/item.ts
import {Subcategory} from "./subcategory";

export class Item {
    private _id: number = 0;
    private _name: string = "";
    private _description: string = "";
    private _price: number = 0;
    private _count: number = 0;
    private _subcategory: Subcategory;
    private _images: string[] = [];
    private _looks: string[] = [];

    constructor(name: string, subcategory: Subcategory) {
        this._name = name;
        this._subcategory = subcategory;
    }

    get id(): number {
        return this._id;
    }

    set id(value: number) {
        this._id = value;
    }

    get name(): string {
        return this._name;
    }

    set name(value: string) {
        this._name = value;
    }

    get description(): string {
        return this._description;
    }

    set description(value: string) {
        this._description = value;
    }

    get price(): number {
        return this._price;
    }

    set price(value: number) {
        this._price = value;
    }

    get count(): number {
        return this._count;
    }

    set count(value: number) {
        this._count = value;
    }

    get subcategory(): Subcategory {
        return <Subcategory>this._subcategory;
    }

    set subcategory(value: Subcategory) {
        this._subcategory = value;
    }

    get images(): string[] {
        return this._images;
    }

    set images(value: string[]) {
        this._images = value;
    }

    get looks(): string[] {
        return this._looks;
    }

    set looks(value: string[]) {
        this._looks = value;
    }
}